cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -204,14 +240,47 @@ pub struct RegisteredEnv {
     pub pointer: EnvironmentPointer,
 }
 
+/// PID of an environment's activation.
+#[derive(Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Pid(u32);
+
+impl Pid {
+    /// Construct a Pid from the current running process.
+    pub fn from_self() -> Self {
+        Pid(nix::unistd::getpid().as_raw() as u32)
+    }
+
+    /// Check whether an activation is still running.
+    fn is_running(&self) -> bool {
+        // TODO: Compare name or check for watchdog child to see if it's a real activation?
+        let pid = NixPid::from_raw(self.0 as i32);
+        match kill(pid, None) {
+            Ok(_) => true,              // known running
+            Err(Errno::EPERM) => true,  // no perms but running
+            Err(Errno::ESRCH) => false, // known not running
```

Comment:
question: What's ESRCH and why do we know its not running?
Maybe amend the comment?

Reply:
yeah i was just missing the "not found" bit, maybe

```suggestion
            Err(Errno::ESRCH) => false, // kill(2) did not find pid: known not running
```

Reply:
Done in https://github.com/flox/flox/pull/1837/commits/381c7032f8295bf2af7d4239e348260a360f3197